src/compiler/utilities.ts

```diff
@@ -6506,7 +6506,9 @@ namespace ts {
         const visited = new Map<string, true>();
         const toCanonical = createGetCanonicalFileName(useCaseSensitiveFileNames);
         for (const basePath of patterns.basePaths) {
-            visitDirectory(basePath, combinePaths(currentDirectory, basePath), depth);
+            if (directoryExists(basePath)) {

```

Comment:
@amcasey I wonder if this check is causing a behavior change in the material-ui benchmark and changing the number of .d.ts files that get loaded or similar.

Reply:
Comparing `--extendedDiagnostics` might be a good first step to figuring out whether there really is an observable behavior change, if you haven't already done that @amcasey

Reply:
Sorry, we'd been discussing this offline and I forgot to update the issue.  The perf suite was using the wrong baseline.  I'll kick off a clean run now, but local testing showed no change in the check time of material-ui.

Reply:
I guess I should also say that I don't believe it will cause a behavioral change because `visitDirectory` would just see an empty list of fs entries for the non-existent directory and stop.

The latest perf run shows no change, as expected. :smile:

Reply:
@amcasey Hi! I don't know if it the right place to ask a question but I found this thread relative to the problem (I'm sorry if it's not). Should `directoryExists` be called with an absolute path like `directoryExists(combinePaths(currentDirectory, basePath))`? 

I came across the undocumented breaking change in 4.4-beta that causes (for example) `ts.sys.readDirectory("", [".ts",".tsx",".d.ts"],["node_modules","dist"],["**/*"])` to return an empty array. Before 4.4-beta it has been returning the correct array of files. So i wonder if it is a bug or desired behaviour.

Thank you in advance.